src/main/java/Program.java

```diff
@@ -1,7 +1,24 @@
-import static Logos.utils.GenerateSql.writeSql;
+import Logos.course.CourseDTO;
+import Logos.course.CourseDao;
+import Logos.subCategory.SubcategoryDao;
+
+import java.sql.SQLException;
+
+import static Logos.utils.GenerateHtml.generateCoursePage;
 
 public class Program {
-    public static void main(String[] args) {
-        writeSql();
+
+    public static void main(String[] args) throws SQLException {
+        generateCoursePage();
+
+        //DAO
+        CourseDao courseDao = new CourseDao();
+
+        SubcategoryDao subcategoryDao = new SubcategoryDao();
+        CourseDTO course = new CourseDTO("Golang", "golang2", 10, "Fábio");
+        courseDao.insert(subcategoryDao.getSubCategoryById(1), course);
+
+        courseDao.delete("golang2");
+        courseDao.turnPublic();
```

Comment:
ao rodar o projeto eu recebo o erro `Cannot invoke "Logos.category.CategoryDTO.getId()" because "this.category" is null` - precisa arrumar. sua classe `GenerateSql` não está sendo mais usada... mas deveria para conseguir gerar as queries de *insert* dentro do *script* que será usado para popular o banco de dados local. (***request changed**)

e aqui novamente vale o que a Madu comentou na classe `CourseDao`, o método `insert()` deveria receber um `course` e não o DTO, e tudo que `course` depende precisa já existir no bando de dados (ou seja, os *inserts* precisam estar no *script* que será gerado para popular o banco). lembre sempre que todo código que entra no github tem que funcionar em qualquer máquina - mesmo que a gente tenha que executar o *script* para popular o banco na mão. (***request changed**)